src/polytopes/chain.jl

```diff
@@ -220,56 +191,35 @@ end
 """
     unique(chain)
 
-Return a new `chain` without duplicate vertices.
+Return a new `chain` without duplicate vertices. Closed chains remain closed.
 """
 Base.unique(c::Chain) = unique!(deepcopy(c))
 
-"""
-    close!(chain)
-
-Close the `chain`, i.e. repeat the first vertex
-at the end of the vertex list.
-"""
-function close!(c::Chain)
-  push!(c.vertices, first(c.vertices))
-  c
-end
-
 """
     close(chain)
 
-Same as [`close!`](@ref) but operates on a copy
-of the original `chain`.
+Close the `chain`, i.e. add a segment going from the last to the first vertex.
 """
-Base.close(c::Chain) = close!(deepcopy(c))
-
-"""
-    open!(chain)
-
-Open the `chain`, i.e. remove the last vertex.
-"""
-function open!(c::Chain)
-  pop!(c.vertices)
-  c
-end
+Base.close(c::Chain) = c
+Base.close(c::Ring) = Chain(CircularVector(c.vertices))
```

Comment:
I think these two should be swapped? If we get a Ring we just return it, otherwise we get a Chain that is open and wrap the vertices in a CircularVector.

Reply:
Done in newer version.